### PR TITLE
BUG: x-axis alpha formatting

### LINKED ISSUE #88: Alpha Correlation X-Axis Number Formatting
Currently the [Alpha Correlation plots](https://view.qiime2.org/visualization/?type=html&src=https%3A%2F%2Fdocs.qiime2.org%2F2017.2%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcm1441%2Fevenness-correlation.qzv) look a little weird:

screen shot 2017-02-12 at 5 21 31 pm

The years are formatted with thousands and decimal place separators. Maybe we do some kind of int test (for decimals) and drop commas (for thousands)?

### PR DESCRIPTION
**Pull request by maxvonhippel**
Formats x axis on alpha correlation visualization to look nice:
1. No commas in thousands, millions, etc.
2. Sensible padding on left and right of data
3. Sensible number of ticks: 
    a. If the x axis data is all integer-valued, then no non-integer ticks on x axis
    b. No more than 12 ticks allowed no matter what on x axis
    c. No less than 3 ticks allowed no matter what on x axis

Fixes #88

### CODE REVIEW

**Comment by maxvonhippel** [comment]
Input:
1. [awesome metadata I wrote](https://forum-qiime2-org.s3-us-west-2.amazonaws.com/original/1X/612bb5f77ee707e41553ddcfd02c1efb58017806.tsv)
2. [artifact to test on](https://forum-qiime2-org.s3-us-west-2.amazonaws.com/original/1X/9642bbb9b3ddde1b390d5b48b132ff3b977fda09.qza)

````BASH
qiime diversity alpha-correlation --i-alpha-diversity evenness_vector.qza --m-metadata-file sample-metadata.tsv --o-visualization groovyviz.qzv
````

Output: [groovy visualization](https://view.qiime2.org/visualization/?type=html&src=https%3A%2F%2Fforum-qiime2-org.s3-us-west-2.amazonaws.com%2Foriginal%2F1X%2Fe5d664e76545403f890b5987cd16b31a266bffa6.qzv)

File: q2_diversity/_alpha/alpha_correlation_assets/src/render.js [hunk 1]
```diff
@@ -17,12 +17,19 @@ export function render(svg, data) {
...
   const { xAxisLabel, yAxisLabel, minX, maxX, minY, maxY } = data;
 
-  const x = scaleLinear().domain([minX - ((maxX - minX) * 0.03), maxX]).range([0, width]).nice();
-  const y = scaleLinear().domain([minY, maxY]).range([height, 0]).nice();
-
   const xAxis = axisBottom();
   const yAxis = axisLeft();
 
+  let pad = (maxX - minX) * 0.03;
+  if (minX % 1 === 0 && maxX % 1 === 0) {
```
Comment (thermokarst) [inline q2_diversity/_alpha/alpha_correlation_assets/src/render.js]: How about we use the JavaScript built-in, [`Number.isInteger()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger) to test if the value is an integer?

File: q2_diversity/_alpha/alpha_correlation_assets/src/render.js [hunk 2]
```diff
@@ -17,12 +17,19 @@ export function render(svg, data) {
...
+    pad = Math.max(Math.round(pad), 1);
+    const between = Math.max(3, Math.round((maxX - minX) + (2 * pad)));
```
Comment (thermokarst) [inline q2_diversity/_alpha/alpha_correlation_assets/src/render.js]: Is it necessary to `round` `(maxX - minX) + (2 * pad)`? Seems like `minX`, `maxX`, and `pad` are already confirmed to be integers at this point.

Comment (maxvonhippel) [inline q2_diversity/_alpha/alpha_correlation_assets/src/render.js]: You're right - I wrote that second round before putting it all in the if clause, and forgot to undo it.  Thanks for pointing this out!

**Comment by maxvonhippel** [comment]
This is ready for another pass @thermokarst !  Thanks!

**Pull request merged**